exp/instancestate/awsinstancestate_controller.go

```diff
@@ -135,7 +136,7 @@ func (r *AwsInstanceStateReconciler) watchQueuesForInstanceEvents() {
 					return
 				}
 				resp, err := sqsSvs.ReceiveMessage(&sqs.ReceiveMessageInput{QueueUrl: aws.String(qp.URL)})
-				if err != nil {
+				if err != nil && !queueNotFoundError(err) {
```

Comment:
This will execute every second so it will flood the logs again if the queue is not created and we are trying to receive msg on that queue.

Reply:
Does `ReceiveMessage()` return `queueNotFoundError`?

Also, it iterates over `r.queueURLs.Range`, so this point is reached only when the queue is created.

Reply:
Yes ReceiveMessage() is also flooding the logs with this error:
controllers/AWSInstanceStateController "msg"="failed to receive messages" "error"="AWS.SimpleQueueService.NonExistentQueue: The specified queue does not exist for this wsdl version.\n\tstatus code: 400, request id: 6b0eae44-fc62-5fb1-ab3f-f46dc2151960"

Reply:
^^this one seems to be a bug, coz even after the queue is created, it's throwing this error

Reply:
@sedefsavas in current runs also I can see these [logs](https://storage.googleapis.com/kubernetes-jenkins/logs/periodic-cluster-api-provider-aws-e2e/1468777921022791680/artifacts/clusters/bootstrap/controllers/capa-controller-manager/capa-controller-manager-7bd56cd58-g4ql6/manager.log)

Reply:
@sedefsavas I have fixed the issue in latest commit

Reply:
Logs are comparatively reduced now but still seeing this error:
> controllers/AWSInstanceStateController "msg"="failed to receive messages" "error"="AWS.SimpleQueueService.NonExistentQueue: The specified queue does not exist for this wsdl version.\n\tstatus code: 400, request id: f4974e87-8498-56fc-968c-14ac7bef669c" 

The problem is when the queue gets deleted, we remove `queueURL` value from the map [here](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/exp/instancestate/awsinstancestate_controller.go#L91) , but the map will still contain the `key` as URL, so [this](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/exp/instancestate/awsinstancestate_controller.go#L129) loop gets executed even if `val` is empty

Reply:
One solution is to execute [this](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/exp/instancestate/awsinstancestate_controller.go#L130) go routine only when val is not empty